#! /usr/bin/env python

PACKAGE='mr_local_planner'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

mode_enum = gen.enum([ gen.const("Stop",           int_t, 0, "A small constant"),
                       gen.const("Demo",           int_t, 1, "A medium constant"),
                       gen.const("Wanderer",       int_t, 2, "A medium constant"),
                       gen.const("Wanderer2",      int_t, 3, "A large constant"),
                       gen.const("WallFollowing",  int_t, 4, "A medium constant"),
                       gen.const("WallFollowing2", int_t, 5, "A large constant"),
                       gen.const("GoTo",           int_t, 6, "A medium constant"),
                       gen.const("GoTo2",          int_t, 7, "A large constant")],
                       "An enum select an operation mode")
gen.add("plot_data",   bool_t,   0, "",  False)
gen.add("mode", int_t, 0, "A size parameter which is edited via an enum", 0, 0, 7, edit_method=mode_enum)
gen.add("map_pix_x", int_t, 0, "map size in pixel", 500, 10, 2000)
gen.add("map_pix_y", int_t, 0, "map size in pixel", 500, 10, 2000)
gen.add("map_max_x", double_t, 0, "map size in meter", 5, -10, 10)
gen.add("map_max_y", double_t, 0, "map size in meter", 5, -10, 10)
gen.add("map_min_x", double_t, 0, "map size in meter",-5, -10, 10)
gen.add("map_min_y", double_t, 0, "map size in meter",-5, -10, 10)
gen.add("map_grid_x", double_t, 0, "grid size in meter", 1, 0, 10)
gen.add("map_grid_y", double_t, 0, "grid size in meter", 1, 0, 10)
gen.add("map_rotation", double_t, 0, "map rotation in rad", 0, -3.14, 3.14)

# my custom parameter for the wanderer1
gen.add("v_mean", double_t, 0, "mean velocity", 0.5, 0.0, 0.8)
gen.add("v_std", double_t, 0, "std dev velocity", 0.1, 0.0, 0.5)
gen.add("w_mean", double_t, 0, "mean rotational velocity", 0.4, 0.0, 0.5)
gen.add("w_std", double_t, 0, "std dev rotational velocity", 0.1, 0.0, 0.5)

exit(gen.generate(PACKAGE, "mr_local_planner", "LocalPlanner"))
